src/sage/schemes/toric/points.py

```diff
@@ -849,7 +849,7 @@ def inhomogeneous_equations(self, ring, nonzero_coordinates, cokernel):
         z = [ring.zero()] * nrays
         for i, value in zip(nonzero_coordinates, z_nonzero):
             z[i] = value
-        return [poly(z) for poly in self.polynomials]
+        return [poly.change_ring(ring)(z) for poly in self.polynomials]
```

Comment:
```suggestion
        newpoly = poly.change_ring(ring)
        return [newpoly(z) for poly in self.polynomials]
```
So we do not create a new polynomial for each item in the list.

Reply:
uh, the comprehension is over `poly`

Reply:
Right, duh. I was thinking the loop was over `z`.